Revision.md

```diff
@@ -0,0 +1,238 @@
+  proof &UCAN
+}
+
+-- When "Commit" transaction is recieved service
+-- performs same steps as with "Patch". Main
+-- difference is that after processing this
+-- transaction Revision will transition to Release
+-- state. If revision was in Release state new state
+-- will contain only provided links, otherwise
+-- it will contain union of all the links that were
+-- received via patches and links provided via
+-- Commit.
+--
+-- General expectation is that service will update IPNS
+-- record after processing "commit" transaction.
+type Commit {
+  -- Revision identifier that is it's public key
+  id ID
+  -- pointer to the head of the pin.
+  head &Release
+  -- The root of the DAG for a new revisions it is
+  -- implicitly implied to be in the links.
+  root Link
+  -- Set of links to be included in the next
+  -- release of the revision.
+  links [Link]
+  -- This would link to the IPLD representation of
+  -- the UCAN (with "commit" capability) in which 
+  -- outermost audience is service ID and innermost
+  -- issuer is pin ID. That way service can verify that
+  -- commit is warranted and generate own IPNS update
+  -- record given it's current state.
+  proof &UCAN
+}
+
+
+
+-- Binary representation of the ed25519 public key
+type ID = Bytes
+-- TODO: Define actual structure
+type UCAN = Link
+```
+
+In the .storage setting it is expected that client will:
+
+1. Provide DAG shard(s) in [CAR][] format.
+4. Include `Transaction` blocks in the provided [CAR][] and list them in [roots](https://ipld.io/specs/transport/car/carv1/#number-of-roots).
+
+In the .storage setting it is expect that service will:
+
+1. Verify that claimed transaction(s) are warrented by provided provided UCAN (Ensuring that client is allowed to update `Revision`).
+3. Perform atomic transaction either succesfully updating ALL `Revisions` (as per transaction) or failing and NOT updating non of the revisions. (Rejecting request and provided [CAR][] all together).
+
+
+### 🚧 Transaction Serialization 🚧
+
+> Note this requires more consideration, what follows is a just a current thinking on the matter.
+
+`Transaction`s MAY be serialized as CAR files. In this serilaziation format transaction to be executed
+will be referenced as CAR roots and point to the DAG-CBOR encoded `Transaction` object with couple of
+nuances:
+
+1. Transaction `links` may link to CAR CID which is to be intepreted as a set of `links` for all the blocks contained in the corresponding CAR.
+
+   > **Note:** We do not currently have CAR IPLD codec. Idea is to iterate on this and define spec based on lessons learned.
+
+2. If `links` field are omitted from transaction object that implies links to all the blocks of this CAR (except transaction blocks).
+
+> 💔 I do not like "implicit links" as that seems impractical in case of multiple transactions. Even for a single transaction case `Transaction` may want to link a DAG known to be available at the destination e.g. CID of the previous revision.
```

Comment:
this is pretty easy to distinguish, because CAR CID’s will have a specific codec, so all other CID’s would be links to previous revision.